cmplxfoil/CMPLXFOIL.py

```diff
@@ -0,0 +1,1120 @@
+            The user can provide an estimate for the lift curve slope in order to accelerate convergence. If the user
+            supply a value to this option, it will not use the delta value anymore to select the angle of attack of the
+            second run. The value should be in 1/deg., by default None
+        maxIter : int, optional
+            Maximum number of iterations, by default 20
+        useNewton : bool, optional
+            If True, Newton's method will be used where the dCL/dAlpha is computed using complex-step, otherwise the
+            secant method is used, by default False
+
+        Returns
+        -------
+        None, but the correct alpha is stored in the aeroProblem
+        """
+        self.setAeroProblem(aeroProblem)
+        if alpha0 is not None:
+            aeroProblem.alpha = alpha0
+        else:
+            alpha0 = aeroProblem.alpha
+
+        if alphaBound is None:
+            alphaBound = (-15, 15)
+        elif isinstance(alphaBound, (int, float)):
+            alphaBound = (-alphaBound, alphaBound)
+        elif isinstance(alphaBound, Iterable):
+            alphaBound = (alphaBound[0], alphaBound[1])
+        else:
+            raise ValueError(
+                f'Supplied alphaBound value "{alphaBound}" is not the correct type, must be a scalar or array-like value.'
+            )
+
+        dCLdAlpha = CLalphaGuess
+        resPrev = None
+        alphaPrev = None
+        hasConverged = False
+        maxRes = 1e2
+
+        for _ in range(maxIter):
+
+            # Call the solver and compute the "residual"
+            self.__call__(aeroProblem, deriv=True)
+            failCheck = {}
+            self.checkSolutionFailure(aeroProblem, failCheck)
+            cl = float(self.funcs[aeroProblem.name]["cl"])
+            print(f"Alpha: {aeroProblem.alpha:>6.3f}, CL: {cl:>7.6f}")
+            res = cl - CLStar
+
+            # Check for convergence and divergence
+            hasConverged = abs(res) < tol
+            if hasConverged:
+                print("Converged!")
+                break
+            elif abs(res > maxRes):
+                break
+
+            # If not converged, compute the next alpha
+            if useNewton:
+                aeroProblem.alpha += 1e-200 * 1j
+                self.__call__(aeroProblem, useComplex=True, deriv=True)
+                dCLdAlpha = np.imag(self.funcsComplex[aeroProblem.name]["cl"]) * 1e200
+                aeroProblem.alpha = np.real(aeroProblem.alpha)
+            else:
+                # If using secant, we can only compute dCLdAlpha from the second iteration onwards
+                if resPrev is not None:
+                    dCLdAlpha = (res - resPrev) / (aeroProblem.alpha - alphaPrev)
+            resPrev = res
+            alphaPrev = aeroProblem.alpha
+
+            # Update the alpha either using dCLdAlpha or delta, or backtracking if something went wrong
+            if dCLdAlpha == 0.0 or failCheck["fail"]:
```

Comment:
I skip the `dCLdAlpha` if the solver fails now, makes sense because we shouldn't trust the CL value that would be used here if the solver fails. I still want to do the convergence/divergence check though, otherwise we might never exit the loop